hyperspy/_signals/eds.py

```diff
@@ -706,7 +706,7 @@ def get_take_off_angle(self):
         tilt_stage = mp.Stage.tilt_alpha
         azimuth_angle = mp.Detector.EDS.azimuth_angle
         elevation_angle = mp.Detector.EDS.elevation_angle
```

Comment:
Catch the error when one of the two attributes is present and raise an error with a more user friendly message, for example, something along the line of:
The take off angle can not be calculated because  the XX angle is not defined in the metadata. Use `set_microscope_parameters` to set it.